ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/StructKeyUtil.java

```diff
@@ -58,16 +57,14 @@ public static KeyBuilder keyBuilder(final ColumnName name, final SqlType type) {
   public static final class KeyBuilder {
 
     private final Schema keySchema;
-    private final org.apache.kafka.connect.data.Field keyField;
 
-    private KeyBuilder(final Schema keySchema) {
+    public KeyBuilder(final Schema keySchema) {
       this.keySchema = Objects.requireNonNull(keySchema, "keySchema");
-      this.keyField = Iterables.getOnlyElement(keySchema.fields());
     }
 
-    public Struct build(final Object keyValue) {
+    public Struct build(final Object keyValue, final int fieldIndex) {
```

Comment:
To check my understanding: this works today because we only ever end up with a single key after partition by or group by, but this method signature will have to change to support building structs with multiple key columns in a future PR, yes?

Reply:
yes - I tried to change this to make it more generic but it causes a shocking amount of code jitter, so I'll do that in a future PR